### ISSUE TITLE
ConvertFromUriLiteral can't work for short data type (Edm.Int16) for function parameter

### ISSUE THREAD

**Issue by VikingsFan** [OData/odata.net#262]
Context is like #154 , uri parser works fine but 

```
ODataUriUtils.ConvertFromUriLiteral(paramValue, ODataVersion.V4, _edmModel, edmParam.Type)
```

throw an exception: "Type verification failed. Expected type 'Edm.Int16' but received the value '1'."

paramValue : string "1"
edmParam.Type : Edm.Int16

Edm.Int32 works fine.

**Comment by congysu** [comment]
@lewischeng-ms and @VikingsFan have a sync.

**Comment by rbeauchamp-lendingtree** [comment]
Same issue for Edm.Byte in OData v4

**Comment by xuzhg** [comment]
Fixed, see #1942
